packages/wallet/src/redux/protocols/direct-funding/readme.md

```diff
@@ -0,0 +1,25 @@
+# Direct Funding Protocol
+
+The direct funding protocol is one of the funding protocols, which means that the protocol progresses the channel from after the exchange of the PreFund setup states to where the participants are able to exchange PostFund states. The keeps track of adjudicator funding in order to track when it is safe to deposit funds.
```

Comment:
I think it might make sense for the direct-funding-protocol to take care of the exchange of the post-fund states too. If you imagine you're the indirect funding protocol, you probably want this taken care of, so the case where you have to directly fund the ledger channel looks more similar to the case where you use an existing ledger channel.

Reply:
:+1:

It seems to me like any funding protocol, on its successful completion, should result in the target channel being in the application phase.

Reply:
I am onboard with that. There is an open question about the implementation of the PostFund state exchange. For indirect funding, we chose to split the protocol into protocolA and protocolB. We could take the same approach for direct funding. Alternatively, we could try out a unified protocol approach.

Reply:
I'm personally for maintaining a single set of states. I'm currently writing the funding protocol, where the states and state machine is nearly identical, and it's tedious and distracting to always remember "Oh yeah, I should make this change in player-b states as well"